bytebuffer-collections/src/main/java/io/druid/collections/spatial/search/PolygonBound.java

```diff
@@ -124,11 +124,28 @@ public boolean contains(float[] coords)
     int j = polyCorners - 1;
     boolean oddNodes = false;
     for (int i = 0; i < polyCorners; i++) {
+      if(abscissa[i] == coords[0] && ordinate[i] == coords[1]) {
+        return true;
+      }
+
+      if(ordinate[i] == ordinate[j] && ordinate[j] == coords[1]) {
+        if ((abscissa[i] < abscissa[j] && abscissa[j] > coords[0] && abscissa[i] < coords[0])
+                || (abscissa[i] > abscissa[j] && abscissa[j] < coords[0] && abscissa[i] > coords[0])) {
+          return true;
+        }
+      }
+
       if ((ordinate[i] < coords[1] && ordinate[j] >= coords[1]
            || ordinate[j] < coords[1] && ordinate[i] >= coords[1])
-          && (abscissa[i] <= coords[0] || abscissa[j] <= coords[0])) {
-        if (abscissa[i] + (coords[1] - ordinate[i]) / (ordinate[j] - ordinate[i]) * (abscissa[j] - abscissa[i])
-            < coords[0]) {
+          && (abscissa[i] <= coords[0] || abscissa[j] <= coords[0]))
+      {
+        if (abscissa[i] + (coords[1] - ordinate[i]) / (ordinate[j] - ordinate[i]) * (abscissa[j] - abscissa[i]) == coords[0])
```

Comment:
Extract `abscissa[i] + (coords[1] - ordinate[i]) / (ordinate[j] - ordinate[i]) * (abscissa[j] - abscissa[i])` expression as a variable and reuse it below